examples/app/flight_delays/main.py

```diff
@@ -0,0 +1,144 @@
+    df1 = airline_delays_df[airline_delays_df.period == date_value]
+
+    if carrier != 'All':
+        df1 = df1[df1.carrier_name == carrier]
+
+    df1 = df1.groupby('airport', as_index=False).sum()
+    df1.set_index('airport', inplace=True)
+
+    if df1.empty:
+        data = {}
+        data['x'] = []
+        data['y'] = []
+        source.data = data
+        return
+
+    df1['late_pct'] = df1['arr_del15'] / df1['arr_flights'] * 100
+    df1 = df1[np.isfinite(df1.late_pct)]
+
+    colors = list(reversed(Reds5))
+    percents_late = df1['late_pct'].tolist()
+    groups = pd.qcut(percents_late, min(len(colors), len(set(percents_late))))
+    df1['color'] = [colors[l] for l in groups.codes]
+
+    size_groups = pd.qcut(df1['arr_flights'].tolist(), len(sizes))
+    df1['size'] = [sizes[l] for l in size_groups.codes]
+
+    df1 = df1.join(airports_df, how='inner')
+    data = {}
+    data['color'] = df1['color']
+    data['x'] = df1['longitude']
+    data['y'] = df1['latitude']
+    data['size'] = df1['size']
+    data['name'] = df1['name']
+    data['late_pct'] = df1['late_pct'].map(lambda r: str(int(r)) + '%')
+    source.data = data
+
+state_xs, state_ys, airline_delays_df, airports_df = load_data()
+year = 2004
+month = 1
+day = 1
+carrier = 'All'
+sizes = [3, 6, 9, 12, 15]
+
+TIPS = [
+    ("Airport", "@name"),
+    ("Delay Rate", "@late_pct"),
+]
+
+source = ColumnDataSource(data=dict(x=[], y=[], color=[], size=[], name=[]))
+
+# create figure
+p = Figure(title="US Airports Delays", plot_width=990, plot_height=570, toolbar_location=None)
+p.patches(state_xs, state_ys, fill_alpha=0.0, line_color="#884444", line_width=2, line_alpha=0.3)
+circle_renderer = p.circle('x', 'y', color='color', size='size', source=source, alpha=.7, line_color='black')
+p.add_tools(HoverTool(renderers=[circle_renderer], tooltips=TIPS))
+p.grid.grid_line_alpha = 0
+
+update_map()
+
+# create controls
+year_slider = Slider(title="Year", value=year, start=2004, end=2014, step=1)
+year_slider.on_change('value', on_year_change)
+
+month_slider = Slider(title="Month", value=month, start=1, end=12, step=1)
+month_slider.on_change('value', on_month_change)
+
+carriers = ['All'] + sorted(airline_delays_df['carrier_name'].unique().tolist())
+airline_selector = Select.create(name='Carrier', value=carrier, options=carriers)
+airline_selector.on_change('value', on_carrier_change)
+
+controls_box = HBox(width=990, children=[year_slider, month_slider, airline_selector])
+
+intro_text = 'The map above shows U.S. airports symbolized by flight delay information for the time period selected above.'
+color_text = 'The color of the airport represents its percentage of delayed flights with lighter red for lower percentage of delays and darker red for greater.'
+size_text = 'The size of the airport represents the total number for flight arrivals.'
+binning_text = 'For both color and size, airports are binned by quintiles.'
```

Comment:
might be nice to put these in a yaml or txt file in the app dir, to demonstrate reading from sidecar resource files.